### PR TITLE
feat: add string type

### PR DESCRIPTION
**Pull request by hongda3141**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-modbus-go/blob/master/.github/CONTRIBUTING.md

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions

  a profile template like :cmd/res/devices/modbus.test.device.string.profile.yml
  if add a string device, you must add a attribute of "wordLength" to tell the length of the device address, 
  if not, default length of a string type device will be 1.

### CODE REVIEW

**Comment by hongda3141** [comment]
how to set up a simulator : not change
set up a device profile : use format like modbus.test.device.string.profile.yml in dir "/cmd/res/example"
create a device : not change the way like before

File: internal/driver/deviceclient.go [hunk 1]
```diff
@@ -50,7 +51,18 @@ func createCommandInfo(req *sdkModel.CommandRequest) (*CommandInfo, error) {
 	if err != nil {
 		return nil, err
 	}
-	length := calculateAddressLength(primaryTable, rawType)
+	var length uint16
+	switch req.Type {
+	case sdkModel.String:
+		l, err := strconv.Atoi(req.Attributes[STR_LENGTH])
```
Comment (weichou1229) [inline internal/driver/deviceclient.go:57]: What is the STR_LENGTH? Is it an address length or string length?

For example, we have a string “Hello World!“ and the **string length** is 12. 
It takes 96 bits (12*8=96), so we should provide 6 register (96/16=6) for storing the  “Hello World!“ string. And the  **address length** is 6.

File: internal/driver/deviceclient.go [hunk 2]
```diff
@@ -177,7 +192,24 @@ func TransformCommandValueToDataBytes(commandInfo *CommandInfo, value *sdkModel.
...
+			less := make([]byte, oribytel-bytesl)
+			dataBytes = append(tempBytes, less...)
+		} else if bytesl > oribytel {
+			dataBytes = tempBytes[:oribytel]
+		} else {
+			dataBytes = []byte(oriStr)
+		}
+	default:
+		dataBytes = value.NumericValue[maxSize-byteCount : maxSize]
+	}
```
Comment (weichou1229) [inline internal/driver/deviceclient.go:212]: Please refactor this code snippet to the if else block https://github.com/edgexfoundry/device-modbus-go/blob/6f937c290bfa4078b9e6fff95cd67f5638b9ced7/internal/driver/deviceclient.go#L220. 
```
if commandInfo.ValueType == sdkModel.Uint32 || commandInfo.ValueType == sdkModel.Int32 || commandInfo.ValueType == sdkModel.Float32 {
		...
} else if commandInfo.ValueType == sdkModel.string{
   ...
}
```

**Review by weichou1229 — CHANGES_REQUESTED** [review]

**Comment by lenny-intel** [comment]
@hongda3141,  Please update the commit message and PR tile per the conventional commit guidelines here:
https://github.com/edgexfoundry/device-modbus-go/blob/master/.github/Contributing.md

Commit bot is currently not running, otherwise it would be failing this PR. 
Working on enabling it.

**Comment by cloudxxx8** [comment]
@hongda3141 could you please reply the change requests?

**Comment by cloudxxx8** [comment]
the code freeze date of this release is 21st June, FYI

**Comment by hongda3141** [comment]
> the code freeze date of this release is 21st June, FYI

sorry for reply so late , i will fix that soon

File: internal/driver/deviceclient.go (hunk 1 above)
Comment (hongda3141) [reply internal/driver/deviceclient.go:57]: ok, i`ll format that

File: internal/driver/deviceclient.go (hunk 2 above)
Comment (hongda3141) [reply internal/driver/deviceclient.go:212]: ok, i`ll reformat there , thank you

**Comment by hongda3141** [comment]
in consideration of commit messages are too much ,maybe create a new branch is better

**Pull request closed without merging**